### ISSUE TITLE
Adding HMR for Hot Reload

### ISSUE THREAD

**Issue by adrianmcli** [brunch/react-skeleton#10]
I'm trying to use the [Brunch HMR API](https://github.com/brunch/hmr-brunch) in order to implement Hot Reload. I essentially followed the steps in [this video](https://www.youtube.com/watch?v=f-ctxG2qEps), which was written for Webpack. But since the Brunch HMR API is similar, I thought it should work.

My [`initialize.js`](https://github.com/adrianmcli/brunch-hmr-example/blob/master/app/initialize.js) is:

```jsx
import ReactDOM from 'react-dom';
import React from 'react';
import App from 'components/App';

document.addEventListener('DOMContentLoaded', () => {
  ReactDOM.render(<App />, document.querySelector('#app'));
});

if (module.hot) {
  module.hot.accept('components/App', (d) => {
    const NextApp = require('components/App').default;
    ReactDOM.render(<NextApp />, document.querySelector('#app'));
  });
}
```

And my [`brunch-config.js`](https://github.com/adrianmcli/brunch-hmr-example/blob/master/brunch-config.js) joins Javascript into only one file:

```js
module.exports = {
  hot: true,
  files: {
    javascripts: {joinTo: 'app.js'},
    stylesheets: {joinTo: 'app.css'}
  },

  plugins: {
    babel: {presets: ['es2015', 'react']}
  }
};
```

After all this, I still get the error:

```
09 Dec 02:22:25 - warn: HMR runtime not found for public/app.js. HMR only works if use compile your JS into a single file, could it be the case?
```

Perhaps I do not understand how HMR really works. Is hot reload possible? Can we do it with this Skeleton?

This is my repository: https://github.com/adrianmcli/brunch-hmr-example

**Comment by denysdovhan** [comment]
Hey @adrianmcli! Thank you for reporting the issue!

I've took a look at your repo. Well, first of all, update your dependencies to the latest versions.

Also, it'd suggest you to take a look at [`with-redux`](https://github.com/brunch/with-redux) skeleton for inspiration. It implements HMR with Redux and React. Here is [a relevant snippet](https://github.com/brunch/with-redux/blob/master/app/initialize.js#L10-L20):

```js
// ...

if (module.hot) {
  module.hot.accept('./reducers', () => {
    store.replaceReducer(require('./reducers').default);
  });
  module.hot.accept();

  module.hot.dispose((data) => {
    data.counter = store.getState();
    [].slice.apply(document.querySelector('#app').children).forEach(function(c) { c.remove() });
  });
}

// ...
```

I think it should help you. Feel free to ask :)

**Comment by kockok** [comment]
I created the project using "with-redux" skeleton.
I got the same error `warn: HMR runtime not found for public/app.js. HMR only works if use compile your JS into a single file, could it be the case?` with [this](https://github.com/brunch/with-redux/blob/master/app/initialize.js). Though I changed the reducers from a **file** to a **folder** like many redux structures out there. 
However, When I completely remove the `if (module.hot)` part, the single file warning still prevail.
